components/SubmissionForm.vue

```diff
@@ -66,24 +72,58 @@
     <span
       class="mb-2 text-neutral-600 text-sm font-normal font-['Noto Sans JP']"
     >{{ $t('submitPage.otherNotes')}}({{ $t('submitPage.optional')}})</span>
-    <input
+    <textarea
     data-testid="submit-input-notes"
       class="mb-[71px] px-3 py-3.5 w-[350px] h-[50px] bg-white rounded-lg border border-zinc-400"
-      type="text"
       v-model="submissionStore.otherNotes"
+      maxlength="300"
     />
     <button
     data-testid="submit-submitbutton"
+    type="submit"
       class="px-20 py-3 rounded-full bg-currentColor w-[350px] text-center text-white text-base font-medium font-['Noto Sans JP']"
-      @click="submissionStore.submit()"
+      @click="validateFields"
     >{{ $t('submitPage.submitButton')}}</button>
-  </div>
+</form>
 </template>
 
   <script lang="ts" setup>
 import { useSubmissionStore } from "~/stores/submissionStore"
 import { useLocaleStore } from "~/stores/localeStore"
+import * as validations from '../utils/formValidations'
+import { computed } from 'vue'
 
 const submissionStore = useSubmissionStore()
 const localeStore = useLocaleStore()
+
+const isValidGoogleMapsUrl: boolean = computed(() => validations.validateGoogleMapsUrlInput(submissionStore.location))
+
+const isValidLastName: boolean = computed(() => validations.validateUserSubmittedLastName(submissionStore.lastName))
+
+const isValidFirstName: boolean = computed(() => validations.validateUserSubmittedFirstName(submissionStore.firstName))
+
+const isValidPrimarySpokenLanguage: boolean = computed(() => validations.validateFirstSpokenLanguage(submissionStore.selectLanguage1))
+
+const isValidSecondarySpokenLanguage: boolean = computed(() => validations.validateSecondSpokenLanguage(submissionStore.selectLanguage2))
+
+const isValidNotes: boolean = computed(() => validations.validateNotes(submissionStore.otherNotes))
```

Comment:
I was getting type errors. So I would add `ComputedRef` type from `vue` to fix the type errors

```suggestion
import type { ComputedRef } from "vue"

const submissionStore = useSubmissionStore()
const localeStore = useLocaleStore()
const isValidGoogleMapsUrl: ComputedRef<boolean> = computed(() => validations.validateGoogleMapsUrlInput(submissionStore.location))
const isValidLastName: ComputedRef<boolean> = computed(() => validations.validateUserSubmittedLastName(submissionStore.lastName))
const isValidFirstName: ComputedRef<boolean> = computed(() => validations.validateUserSubmittedFirstName(submissionStore.firstName))
const isValidPrimarySpokenLanguage: ComputedRef<boolean> = computed(() => validations.validateFirstSpokenLanguage(submissionStore.selectLanguage1))
const isValidSecondarySpokenLanguage: ComputedRef<boolean> = computed(() => validations.validateSecondSpokenLanguage(submissionStore.selectLanguage2))
const isValidNotes: ComputedRef<boolean> = computed(() => validations.validateNotes(submissionStore.otherNotes))
```

Reply:
➕